LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -520,6 +545,25 @@ internal async Task<bool> UpdateAfterJoinAsync(string devAddr, string nwkSKey, s
             reportedProperties[TwinProperty.NetID] = netID;
             reportedProperties[TwinProperty.DevNonce] = devNonce;
 
+            // Additional Join Property Saved
+            if (this.DesiredRX1DROffset != DefaultJoinValues)
+            {
+                reportedProperties[TwinProperty.RX1DROffset] = this.DesiredRX1DROffset;
+            }
+            else
+            {
+                reportedProperties[TwinProperty.RX1DROffset] = null;
+            }
+
+            if (this.DesiredRX2DataRate != DefaultJoinValues)
+            {
+                reportedProperties[TwinProperty.RX2DataRate] = this.DesiredRX2DataRate;
+            }
+            else
+            {
+                reportedProperties[TwinProperty.RX2DataRate] = null;
```

Comment:
why do you blank it out if they are the same value? Wouldn't we want to keep the value to ensure the same behavior will apply next time this runs?

Reply:
Thinking behind I suppose is to save twin size. Not having a property or having with default value results in the same.

Reply:
setting it to null basically deletes this twin property. Not sure that is the intention here but this would be the effect

Reply:
Yes, this was the intended behavior